### PR TITLE
Split Data in to logical modules

### PR DESCRIPTION
**Pull request by charleso**
/cc @nhibberd @tmcgilchrist 

_Nothing_ changed here, just shuffling code. As always, module naming suggestions welcome.

### CODE REVIEW

File: src/Ambiata/Cli/Data/Env.hs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+import           Ambiata.Cli.Data.Api
+
+import           P
+
+
+data RunMode = Daemon | OneShot
+  deriving (Eq, Show)
+
+data Verbosity = Verbose | Quiet
```
Comment (nhibberd) [inline src/Ambiata/Cli/Data/Env.hs]: Is this needed in the src tree?

File: src/Ambiata/Cli/Data/Download.hs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import           P
+
+import           System.IO (FilePath)
+
+
+newtype DownloadDir = DownloadDir {
+  unDownloadDir :: FilePath
+} deriving (Show, Eq, Ord)
+
+newtype DownloadResult = DownloadResult { unDownloadResult :: [LocalFile] }
```
Comment (nhibberd) [inline src/Ambiata/Cli/Data/Download.hs]: I know your probably just shifting things around here. As a note though, some consistent formatting would be good, see `DownloadDir` vs `DownloadResult`

File: src/Ambiata/Cli/Data/Transfer.hs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module Ambiata.Cli.Data.Transfer (
```
Comment (nhibberd) [inline src/Ambiata/Cli/Data/Transfer.hs]: Might be tempted to name this `Credentials`

**Comment by nhibberd** [comment]
:cat:

File: src/Ambiata/Cli/Data/Download.hs (hunk 2 above)
Comment (charleso) [inline src/Ambiata/Cli/Data/Download.hs]: > I know your probably just shifting things around here

Yeah I am, and I tried to leave things to be identical. I might do another quick PR with formatting.

**Pull request merged**